server/src/services/asset.service.ts

```diff
@@ -328,4 +328,15 @@ export class AssetService {
       await this.jobRepository.queue({ name: JobName.SIDECAR_WRITE, data: { id, ...writes } });
     }
   }
+
+  async getUniqueOriginalPaths(auth: AuthDto): Promise<string[]> {
+    return this.assetRepository.getUniqueOriginalPaths(auth.user.id);
+  }
+
+  async getAssetsByOriginalPath(auth: AuthDto, path: string): Promise<AssetResponseDto[]> {
+    const assets = await this.assetRepository.getAssetsByOriginalPath(auth.user.id, path);
+    const ids = assets.map((a) => a.id);
+    await this.access.requirePermission(auth, Permission.ASSET_READ, ids);
```

Comment:
If the query itself is filtered by the user id, I don't see any reason to do an access check.